tests/device_tests/test_nonstandard_paths.py

```diff
@@ -0,0 +1,255 @@
+# This file is part of the Trezor project.
+#
+# Copyright (C) 2012-2021 SatoshiLabs and contributors
+#
+# This library is free software: you can redistribute it and/or modify
+# it under the terms of the GNU Lesser General Public License version 3
+# as published by the Free Software Foundation.
+#
+# This library is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU Lesser General Public License for more details.
+#
+# You should have received a copy of the License along with this library.
+# If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.
+
+import pytest
+
+from trezorlib import btc, messages
+from trezorlib.tools import parse_path
+
+from ..tx_cache import TxCache
+
+TX_CACHE_MAINNET = TxCache("Bitcoin")
+
+TXHASH_6189e3 = bytes.fromhex(
+    "6189e3febb5a21cee8b725aa1ef04ffce7e609448446d3a8d6f483c634ef5315"
+)
+TXHASH_d5f65e = bytes.fromhex(
+    "d5f65ee80147b4bcc70b75e4bbf2d7382021b871bd8867ef8fa525ef50864882"
+)
+
+
+VECTORS = (
+    # GreenAddress A m/[1,4]/address_index
+    (
+        "m/4/255",
+        (
+            messages.InputScriptType.SPENDADDRESS,
+            messages.InputScriptType.SPENDWITNESS,
+            messages.InputScriptType.SPENDP2SHWITNESS,
+        ),
+    ),
+    # GreenAddress B m/3'/[1-100]'/[1,4]/address_index
+    (
+        "m/3'/100'/4/255",
+        (
+            messages.InputScriptType.SPENDADDRESS,
+            messages.InputScriptType.SPENDWITNESS,
+            messages.InputScriptType.SPENDP2SHWITNESS,
+        ),
+    ),
+    # GreenAdress Sign A m/1195487518
+    (
+        "m/1195487518",
+        (
+            messages.InputScriptType.SPENDADDRESS,
+            messages.InputScriptType.SPENDWITNESS,
+            messages.InputScriptType.SPENDP2SHWITNESS,
+        ),
+    ),
+    # GreenAdress Sign B m/1195487518/6/address_index
+    (
+        "m/1195487518/6/255",
+        (
+            messages.InputScriptType.SPENDADDRESS,
+            messages.InputScriptType.SPENDWITNESS,
+            messages.InputScriptType.SPENDP2SHWITNESS,
+        ),
+    ),
+    # Casa m/49/coin_type/account/change/address_index
+    (
+        "m/49/0/63/0/255",
+        (messages.InputScriptType.SPENDP2SHWITNESS,),
+    ),
+    # Unchained hardened m/45'/coin_type'/account'/[0-1000000]/change/address_index
```

Comment:
are you sure UC paths apply for single-sig?

Reply:
Don't know. I copied whatever `validate_path_against_script_type` allows in `apps/bitcoin/keychain.py`.

Reply:
ah, but that branch only applies if `multisig` is set, because in that case `SPENDADDRESS` is equivalent to `SPENDMULTISIG`

this btw maybe indicates a problem either in keychain or in the test -- these paths should not be allowed for single-sig

Reply:
Uh, right, should be fixed in 1102ebdddbc2158b1f950045e4b6db04e2816a60.

Reply:
ISTM that failing to validate path against script type only results in warning dialog for `get_address` as well as `sign_tx`.